src/optimagic/optimization/create_optimization_problem.py

```diff
@@ -161,6 +162,14 @@ def create_optimization_problem(
         )
         raise MissingInputError(msg)
 
+    if fun_and_jac is not None and fun is None:
```

Comment:
```suggestion
    if fun_and_jac is not None and fun is None and criterion is None:
```

We need the check for `criterion` until the deprecation of `criterion` is complete.